packages/plugin-conversation/src/conversation.js

```diff
@@ -101,6 +124,28 @@ const Conversation = SparkPlugin.extend({
       .then(() => res.body));
   },
 
+  leave(conversation, object, activity) {
+    this._inferConversationUrl(conversation);
```

Comment:
It seems like most places we handle the promise `_inferConversationUrl()` returns and treat it as though it's async despite the fact it's just a sync function reading something off the device. Even though the whole flow for this function is all synchronous and it *works* this way I think we should keep following the promise pattern. Thoughts?